### PR TITLE
[SHELL32] Prevent a second call to  Drive Properties Dialog

### PR DESCRIPTION
**Pull request by zecarlos1957**
Clicking on the Properties item in the drives context menu will open two Drives Properties dialogs one after the other.

CDefaultContextMenu::DoProperties provides a fallback call to property sheet testing the return value of the _DoCallback method, which is ultimately the return value of SH_ShowDriveProperties(...).
_Do Callback returns 0 (on error) and 1(on success), S_FALSE is defined as 1. 

https://jira.reactos.org/browse/CORE-18613

### CODE REVIEW

**Comment by KRosUser2** [comment]
Link to Jira ticket is missing

File: dll/win32/shell32/CDefaultContextMenu.cpp [hunk 1]
```diff
@@ -842,7 +842,7 @@ CDefaultContextMenu::DoProperties(
     HRESULT hr = _DoCallback(DFM_INVOKECOMMAND, DFM_CMD_PROPERTIES, NULL);
 
     // We are asked to run the default property sheet
-    if (hr == S_FALSE)
+    if (hr == FALSE)
```
Comment (learn-more) [inline dll/win32/shell32/CDefaultContextMenu.cpp:845]: `_DoCallback` returns an `HRESULT`, and `FALSE` is not an `HRESULT` so this is not a fix but a hack.

**Review by learn-more — CHANGES_REQUESTED** [review]
This is not a fix, but rather a workaround or a hack.

File: dll/win32/shell32/CDefaultContextMenu.cpp (hunk 1 above)
Comment (hpoussin) [reply dll/win32/shell32/CDefaultContextMenu.cpp:845]: `SH_ShowDriveProperties` is marked as returning a `BOOL`, but sometimes returns a `HRESULT`.
Then, `DrivesContextMenuCallback` handles this result as an `HRESULT`.
This needs to be fixed.

File: dll/win32/shell32/folders/CDrivesFolder.cpp [hunk 2]
```diff
@@ -345,7 +345,7 @@ HRESULT CALLBACK DrivesContextMenuCallback(IShellFolder *psf,
         {
             // pdtobj should be valid at this point!
             ATLASSERT(pdtobj);
-            hr = SH_ShowDriveProperties(wszBuf, pdtobj);
+            hr = SH_ShowDriveProperties(wszBuf, pdtobj) ? 0 : (HRESULT)-1;
```
Comment (learn-more) [inline dll/win32/shell32/folders/CDrivesFolder.cpp:348]:
```suggestion
            hr = SH_ShowDriveProperties(wszBuf, pdtobj) ? `S_OK` : `E_UNEXPECTED`;
```
Alternatively, make `SH_ShowDriveProperties` return an `HRESULT`

**Review by learn-more — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
I have tested this PR, but according to the test results, it **does not** fix [CORE-18613](https://jira.reactos.org/browse/CORE-18613), but fixes [CORE-18537](https://jira.reactos.org/browse/CORE-18537) instead. :smile:

**Review by HBelusca — APPROVED** [review]

**Pull request merged**